File: Go/bot.go
package main

import (
  "time"

  "github.com/hybridgroup/gobot"
  "github.com/hybridgroup/gobot/platforms/firmata"
  "github.com/hybridgroup/gobot/platforms/gpio"
)

const PWM_L = "10"
const PWM_R = "9"
const DIR_L = "8"
const DIR_R = "7"

func main() {
  gbot := gobot.NewGobot()
  firmataAdaptor := firmata.NewFirmataAdaptor("arduino", "/dev/cu.usbmodem1421")

  leftMotor := newMotor(firmataAdaptor, "left", PWM_L, DIR_L)
  rightMotor := newMotor(firmataAdaptor, "right", PWM_R, DIR_R)

  work := func() {
    speed := byte(50);
    fwd := true

    gobot.Every(1*time.Second, func() {
      if (fwd) {
        leftMotor.Forward(speed)
        rightMotor.Forward(speed)
      } else {
        leftMotor.Backward(speed)
        rightMotor.Backward(speed)
      }

      fwd = !fwd
    })
  }

  robot := gobot.NewRobot("bot",
    []gobot.Connection{firmataAdaptor},
    []gobot.Device{leftMotor,rightMotor},
    work,
  )

  gbot.AddRobot(robot)

  gbot.Start()
}

func newMotor(adapter gpio.DigitalWriter, name string, speedPin string, directionPin string) *gpio.MotorDriver {
  motor := gpio.NewMotorDriver(adapter, name, speedPin)
  motor.DirectionPin = directionPin

  return motor
}

File: Ruby/bot.rb
require 'artoo'

connection :arduino, adaptor: :firmata, port: '/dev/cu.usbmodem1421'
device :left_motor, :driver => :motor, :speed_pin => 10, forward_pin: 13, backward_pin: 8
device :right_motor, :driver => :motor, :speed_pin => 9, forward_pin: 13, backward_pin: 7

work do
  fwd = true
  speed = 50

  every(1.seconds) do
    if (fwd)
      left_motor.forward speed
      right_motor.forward speed
    else
      left_motor.backward speed
      right_motor.backward speed
    end

    fwd = !fwd
  end
end
